tests/Microsoft.Bot.Connector.Tests/OAuthConnectorTests.cs

```diff
@@ -59,45 +86,241 @@ public async Task GetUserToken_ShouldReturnNullOnInvalidConnectionString()
         }
 
         [Fact]
-        public async Task SignOutUser_ShouldThrowOnEmptyUserId()
+        public async Task SignOutWithHttpMessagesAsync_ShouldThrowOnEmptyUserId()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            await Assert.ThrowsAsync<ValidationException>(() => client.UserToken.SignOutAsync(null, "dummyConnection"));
+        }
+
+        [Fact]
+        public async Task SignOutWithHttpMessagesAsync_ShouldThrowOnNoLocalBot()
         {
             var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
-            await Assert.ThrowsAsync<ValidationException>(() => client.UserToken.SignOutAsync(null, "mockConnection"));
+            ServiceClientTracing.IsEnabled = true;
+
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_OS")) &&
+                Environment.GetEnvironmentVariable("AGENT_OS").Equals("Windows_NT"))
+            {
+                // Automated Windows build exception:
+                await Assert.ThrowsAsync<ErrorResponseException>(() => client.UserToken.SignOutAsync(
+                    "dummyUserId", "dummyConnectionName", "dummyChannelId"));
+            }
+            else
+            {
+                // MacLinux build and local build exception:
+                await Assert.ThrowsAsync<System.Net.Http.HttpRequestException>(() => client.UserToken.SignOutAsync(
+                    "dummyUserId", "dummyConnectionName", "dummyChannelId"));
+            }
```

Comment:
Is it possible to refactor this to avoid code repetition?

Reply:
I don't see how. The calls and the exceptions tested for are different. And to replace 2 lines of setup code with a call to a helper routine seems to me to be an increase of complexity rather than a reduction.